FantasyUniverse/Services/Auth/FirebaseAuthService.swift

```diff
@@ -30,97 +36,159 @@ final class FirebaseAuthService: ObservableObject {
         state = .proccessing
         handle = Auth
             .auth()
-            .addStateDidChangeListener { (_, user) in
+            .addStateDidChangeListener { [weak self] (_, user) in
+                guard let self = self else { return }
                 if let user = user {
-                    print("Got user: \(user)")
                     
-                   let user = User(
-                        uid: user.uid,
-                        displayName: user.displayName,
-                        email: user.email,
-                        refreshToken: user.refreshToken)
-                    self.state = .success(with: user)
+                    self.fetchedUserInfoPublisher(uid: user.uid)
```

Comment:
To me, looks like a perfect opportunity to do publisher chaining with the `flatMap` :D